### PR TITLE
New addons base image

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Move to a new clear centos 7 base image.

Still to do:

* do we need peep-support? (which was available in the old base-images but we don't seem to use it?)

This requires nodejs 0.12.x https://bugzilla.mozilla.org/show_bug.cgi?id=1226533

### CODE REVIEW

**Comment by muffinresearch** [comment]
> Do we need mozilla/marketplace-env@73e58cc

With a base image running Centos 7 you probably want to check what's the the main repos. It would also be worth checking with jason to see what we're running in prod.

I'm hopeful a lot of that will be not needed.

**Comment by muffinresearch** [comment]
Seems a little odd to me to make this a PR. Questions might be better placed in the issues as an alternative.

I see the "not-ready" but the title was confusing. I was thinking "we don't want to do that" :). But I see this is just for you and not something you'd intend to merge.

**Comment by EnTeQuAk** [comment]
Yeah, I did this mostly just to have a reference for stuff that is code-related. Some of this stuff might go to the addons base image too but I found it nicer to discuss about that in a PR rather than an issue with no code references.

**Comment by EnTeQuAk** [comment]
Oh and I totally messed up the title, auto-filled titles from git commits make me oversee that every time :disappointed:

File: docker-compose.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 worker: &worker
-  image: addons/olympia
+  build: .
```
Comment (EnTeQuAk) [inline docker-compose.yml]: As a side note: This is only for development, once this PR is r+ the image will default to addons/olympia again.

**Comment by EnTeQuAk** [comment]
@muffinresearch @magopian I'd love to have some feedback on this.

Actually now that I'm thinking about it - almost all of the stuff from this PR should go in the addons/olympia docker file right?

**Comment by EnTeQuAk** [comment]
r?

File: Dockerfile [hunk 2]
```diff
@@ -20,15 +20,22 @@ RUN yum update -y \
...
+        python-devel \
+        nodejs \
     && yum clean all
 
-# The version in the above image is ancient, and does not support the
-# --no-binary flag used in our requirements files.
-# We also need to install wheels.
-RUN pip install -U pip wheel
+RUN curl https://raw.githubusercontent.com/pypa/pip/master/contrib/get-pip.py > get-pip.py \
+    && python get-pip.py
```
Comment (magopian) [inline Dockerfile]: isn't pip available anymore?

**Comment by magopian** [comment]
I'm not very knowledgeable regarding docker, but what I see seems to make sense.

File: docker/mysql.repo [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[mysql56-community]
```
Comment (muffinresearch) [inline docker/mysql.repo]: Do we still need this. What version is in the main repos and how does that tie up with prod. What does prod use?

File: Dockerfile [hunk 4]
```diff
@@ -20,15 +20,22 @@ RUN yum update -y \
...
+        openssl-devel \
+        python-devel \
+        nodejs \
     && yum clean all
 
-# The version in the above image is ancient, and does not support the
-# --no-binary flag used in our requirements files.
-# We also need to install wheels.
-RUN pip install -U pip wheel
+RUN curl https://raw.githubusercontent.com/pypa/pip/master/contrib/get-pip.py > get-pip.py \
```
Comment (muffinresearch) [inline Dockerfile]: Is there not a system package for this?

File: docker/mysql.repo [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+name=MySQL 5.6 Community Server
+baseurl=http://repo.mysql.com/yum/mysql-5.6-community/el/7/$basearch/
```
Comment (muffinresearch) [inline docker/mysql.repo]: If we do retain this can this be available over https?

File: docker/mysql.repo [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+enabled=1
+gpgcheck=0
```
Comment (muffinresearch) [inline docker/mysql.repo]: Would be nice to import the necessary keys here so we don't have to turn this off.

File: docker/nodesource.repo [hunk 7]
```diff
@@ -0,0 +1,6 @@
+[nodesource]
+name=Node.js Packages for Enterprise Linux 7 - $basearch
+baseurl=https://rpm.nodesource.com/pub_0.12/el/7/$basearch
+failovermethod=priority
+enabled=1
+gpgcheck=0
```
Comment (muffinresearch) [inline docker/nodesource.repo]: As above

File: Dockerfile [hunk 8]
```diff
@@ -20,15 +20,22 @@ RUN yum update -y \
         cyrus-sasl-devel \
         libxml2-devel \
         libxslt-devel \
-        nodejs \
         zlib-devel \
-        mysql-community-libs-compat-5.6.14-3.el6.x86_64 \
+        git \
```
Comment (muffinresearch) [inline Dockerfile]: What's the git dep needed for?

File: Dockerfile (hunk 2 above)
Comment (EnTeQuAk) [inline Dockerfile]: Not in centos, never was, in the old image we're installing it manually too.

File: Dockerfile (hunk 8 above)
Comment (EnTeQuAk) [inline Dockerfile]: For verything that we're aquiring from github in https://github.com/mozilla/olympia/blob/master/requirements/prod.txt#L95

File: docker/mysql.repo (hunk 6 above)
Comment (EnTeQuAk) [inline docker/mysql.repo]: Good point, I'll do that.

File: docker/mysql.repo (hunk 5 above)
Comment (EnTeQuAk) [inline docker/mysql.repo]: absolutely.

File: Dockerfile (hunk 8 above)
Comment (muffinresearch) [inline Dockerfile]: Ah ok - makes sense. Maybe add a comment to that effect?

**Comment by EnTeQuAk** [comment]
@muffinresearch r? I updated the pyrepo references so that we're now prefering the same packages we use on prod (e.g fixes mysql-python linking issue) and we're now verifying the gpg keys of all source repos.

File: Dockerfile [hunk 9]
```diff
@@ -20,20 +26,35 @@ RUN yum update -y \
...
+        mysql-community-client \
+        mysql-community-libs \
     && yum clean all
 
-# The version in the above image is ancient, and does not support the
-# --no-binary flag used in our requirements files.
-# We also need to install wheels.
-RUN pip install -U pip wheel
+# Unfortunately it seems pip isn't available in the centos repository.
+RUN curl https://raw.githubusercontent.com/pypa/pip/master/contrib/get-pip.py > get-pip.py \
```
Comment (muffinresearch) [inline Dockerfile]: What do ops do to get pip? 

Looks like it's in EPEL (Extra Packages for Enterprise Linux) see https://fedoraproject.org/wiki/EPEL
You can install that repo with the `epel-release` package and then pip  is installable as python-pip.

Then you should be able to install `python-pip` from yum.

**Comment by muffinresearch** [comment]
I think there's some open questions on what ops are using for sources (e.g. where does their `pip` and `mysql` deps come from) but that can be dealt with later. I also wonder could we use a system package for `mysql-python`? Presumably that would save on the mysql deps in the first place?

For node I see they are mirroring nodesource so the solution here should be equivalent for all intents and purposes. Which is great. :tada: 

I think it would be nicer to use EPEL for python-pip vs curling a python script. I added instructions for that in the comments above.

Fix that and assuming olympia runs fine on this - I think this is fine to land and qualifying the sources can be a follow-up. 

__Don't forget to remove the local build line.__ :+1: 

r+wc

**Pull request merged**